num/src/bigint.rs

```diff
@@ -848,6 +848,54 @@ macro_rules! construct_bigint {
             }
         }
 
+        impl core::fmt::UpperHex for $name {
+            fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> Result<(), core::fmt::Error> {
+                if f.alternate() {
+                    write!(f, "0x")?;
+                }
+                for chunk in self.0.iter().rev() {
+                    write!(f, "{:0>16X}", chunk)?;
```

Comment:
I think we need to follow formatting patters from `f`, i.e. the width and alignment instructions. This probably can be done with something like
`"{:0$widthX}", $value, width=f.width()`

Reply:
Makes sense, I will add them